src/datasets/download/download_config.py

```diff
@@ -93,7 +93,7 @@ def __post_init__(self, use_auth_token):
                 FutureWarning,
             )
             self.token = use_auth_token
-        if "hf" not in self.storage_options:
+        if self.token is not None and "hf" not in self.storage_options:
```

Comment:
I was worried that removing this altogether might break functionality for someone. This still might break CI but let's see before spending more time on it.

Reply:
you might have to revert this, since we need the endpoint to be in the storage options even if token is None

Reply:
I'm trying to account for that [here](https://github.com/huggingface/datasets/pull/6862/files#diff-b25bef23b0bcf8e3bb59f625d727e585236772d449eae9dee5d32def98ffd321R201). This way `_prepare_path_and_storage_options` ensures that the endpoint is in the storage options without needing to populate it for all protocols in `__post_init__`.

Reply:
I think we can completely remove the default "hf" storage option from `DownloadConfig`, now that it is properly handled by the call to `_prepare_path_and_storage_options` made in `cached_path`.

Reply:
Do you agree, @lhoestq?

Reply:
yes sounds good :)